sympy/polys/tests/test_rootoftools.py

```diff
@@ -244,7 +250,23 @@ def test_CRootOf_evalf():
 
     # watch out for UnboundLocalError
     c = CRootOf(90720*x**6 - 4032*x**4 + 84*x**2 - 1, 0)
-    assert str(c._eval_evalf(2)) == '-0.e-1'
+    assert c._eval_evalf(2)  # doesn't fail
+
+    # watch out for imaginary parts that don't want to evaluate
+    assert str(RootOf(x**16 + 32*x**14 + 508*x**12 + 5440*x**10 +
+        39510*x**8 + 204320*x**6 + 755548*x**4 + 1434496*x**2 +
+        877969, 10).n(2)) == '-3.4*I'
+
+    # check reset
+    r = [RootOf(x**3 + x + 3, i) for i in range(3)]
+    r[0]._reset()
+    for ri in r:
+        i = ri._get_interval()
+        n = ri.n(2)
+        assert str(i) != str(ri._get_interval())
+        ri._reset()
+        # XXX using strings because intervals don't compare equal
```

Comment:
Could `__eq__` be redefined to satisfy this?

Reply:
I can try. I'm not sure why the equality fails as it is.